src/rust/engine/fs/store/src/snapshot_ops.rs

```diff
@@ -684,6 +690,7 @@ pub trait SnapshotOps: StoreWrapper + 'static {
     &self,
     root_digest: Digest,
     prefix: RelativePath,
+    extra_paths_behavior: ExtraPathsBehavior,
```

Comment:
Another way to do this would be to use `SnapshotOps::subset` to select only the relevant directory, and to then strip the prefix from that directory.

Reply:
Happy to do either - do you have a strong preference either way?

Reply:
It feels like there might be a slight clarity benefit to `subset`+`strip` because it doesn't involve new arguments (could mention using `subset` in the error message even?), but no: no preference.

Reply:
Done :)